src/test/scala/org/ergoplatform/nodeView/wallet/scanning/ScanningPredicateJsonCodecsSpecification.scala

```diff
@@ -0,0 +1,54 @@
+package org.ergoplatform.nodeView.wallet.scanning
+
+import io.circe.parser._
+import org.ergoplatform.ErgoBox
+import org.ergoplatform.utils.ErgoPropertyTest
+import org.ergoplatform.utils.generators.ErgoTransactionGenerators
+import scorex.crypto.hash.Digest32
+import scorex.util.encode.Base16
+
+class ScanningPredicateJsonCodecsSpecification extends ErgoPropertyTest with ErgoTransactionGenerators {
+
+  import ScanningPredicateJsonCodecs.{scanningPredicateEncoder, scanningPredicateDecoder}
+
+  property("complex or roundtrip") {
+    val p = ScanningPredicateTestVectors.complexOr
```

Comment:
Let's also use generators.

Reply:
fixed in a982f5ce7b70f71357cc4ac847e12be3847d788f